subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/OffsetRelativePath.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.snapshot;
+
+import java.util.function.IntUnaryOperator;
+
+import static org.gradle.internal.snapshot.CaseSensitivity.CASE_SENSITIVE;
+import static org.gradle.internal.snapshot.PathUtil.compareChars;
+import static org.gradle.internal.snapshot.PathUtil.compareCharsIgnoringCase;
+import static org.gradle.internal.snapshot.PathUtil.equalChars;
+import static org.gradle.internal.snapshot.PathUtil.isFileSeparator;
+
+public class OffsetRelativePath {
```

Comment:
I don't think we should have `offset` in the name. The point of introducing the class is to hide that kind of implementation detail.

Reply:
So you would suggest calling this `RelativePath`? Or do you have a different name in mind?

I currently wouldn't try to hide the implementation detail, given that the other relative path we are using throughout the code is a `String`, so the this implementation of the (non-existing) `RelativePath` interface is based on an offset and an absolute path.

Reply:
`PathSuffix` ?